core/src/main/java/software/amazon/awssdk/core/auth/InstanceProfileCredentialsProvider.java

```diff
@@ -17,37 +17,25 @@
 
 import java.io.IOException;
 import java.net.URI;
-import java.net.URISyntaxException;
+import software.amazon.awssdk.core.AwsSystemSetting;
 import software.amazon.awssdk.core.SdkClientException;
 import software.amazon.awssdk.core.internal.CredentialsEndpointProvider;
-import software.amazon.awssdk.core.internal.EC2CredentialsUtils;
-import software.amazon.awssdk.core.util.EC2MetadataUtils;
-import software.amazon.awssdk.utils.SdkAutoCloseable;
+import software.amazon.awssdk.core.internal.HttpCredentialsUtils;
 
 /**
  * Credentials provider implementation that loads credentials from the Amazon EC2 Instance Metadata Service.
  */
-public class InstanceProfileCredentialsProvider implements AwsCredentialsProvider, SdkAutoCloseable {
-    /**
-     * The client to use to fetch the Amazon ECS credentials.
-     */
-    private final EC2CredentialsProvider credentialsFetcher;
+public class InstanceProfileCredentialsProvider extends HttpCredentialsProvider {
+
+    //TODO: make this private
+    static final String SECURITY_CREDENTIALS_RESOURCE = "/latest/meta-data/iam/security-credentials/";
```

Comment:
Should this be a reviewbeforerelease? What do we need to do to make this private?

Reply:
we should probably port the custom mock -service to wire-mock (This is exposed so that that thing can use it, it's package private so i'm not *overly* worried. I think i'll just remove the TODO for now

Reply:
Cool, we could always mark it as a test API.